actors/market/tests/harness.rs

```diff
@@ -0,0 +1,750 @@
+    rt.set_value(amount.clone());
+
+    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, addr);
+
+    rt.expect_validate_caller_type(vec![*ACCOUNT_ACTOR_CODE_ID, *MULTISIG_ACTOR_CODE_ID]);
+
+    assert!(rt
+        .call::<MarketActor>(Method::AddBalance as u64, &RawBytes::serialize(addr).unwrap(),)
+        .is_ok());
+
+    rt.verify();
+
+    rt.add_balance(amount);
+}
+
+pub fn withdraw_provider_balance(
+    rt: &mut MockRuntime,
+    withdraw_amount: TokenAmount,
+    expected_send: TokenAmount,
+    provider: Address,
+    owner: Address,
+    worker: Address,
+) {
+    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, worker);
+    rt.expect_validate_caller_addr(vec![owner, worker]);
+    expect_provider_control_address(rt, provider, owner, worker);
+
+    let params = WithdrawBalanceParams { provider_or_client: provider, amount: withdraw_amount };
+
+    rt.expect_send(
+        owner,
+        METHOD_SEND,
+        RawBytes::default(),
+        expected_send.clone(),
+        RawBytes::default(),
+        ExitCode::OK,
+    );
+    let ret: WithdrawBalanceReturn = rt
+        .call::<MarketActor>(Method::WithdrawBalance as u64, &RawBytes::serialize(params).unwrap())
+        .unwrap()
+        .deserialize()
+        .unwrap();
+    rt.verify();
+
+    assert_eq!(
+        expected_send, ret.amount_withdrawn,
+        "return value indicates {} withdrawn but expected {}",
+        ret.amount_withdrawn, expected_send
+    );
+}
+
+pub fn withdraw_client_balance(
+    rt: &mut MockRuntime,
+    withdraw_amount: TokenAmount,
+    expected_send: TokenAmount,
+    client: Address,
+) {
+    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, client);
+    rt.expect_send(
+        client,
+        METHOD_SEND,
+        RawBytes::default(),
+        expected_send.clone(),
+        RawBytes::default(),
+        ExitCode::OK,
+    );
+    rt.expect_validate_caller_addr(vec![client]);
+
+    let params = WithdrawBalanceParams { provider_or_client: client, amount: withdraw_amount };
+
+    let ret: WithdrawBalanceReturn = rt
+        .call::<MarketActor>(Method::WithdrawBalance as u64, &RawBytes::serialize(params).unwrap())
+        .unwrap()
+        .deserialize()
+        .unwrap();
+    rt.verify();
+
+    assert_eq!(
+        expected_send, ret.amount_withdrawn,
+        "return value indicates {} withdrawn but expected {}",
+        ret.amount_withdrawn, expected_send
+    );
+}
+
+pub fn activate_deals(
+    rt: &mut MockRuntime,
+    sector_expiry: ChainEpoch,
+    provider: Address,
+    current_epoch: ChainEpoch,
+    deal_ids: &[DealID],
+) {
+    rt.set_caller(*MINER_ACTOR_CODE_ID, provider);
+    rt.expect_validate_caller_type(vec![*MINER_ACTOR_CODE_ID]);
+
+    let params = ActivateDealsParams { deal_ids: deal_ids.to_vec(), sector_expiry };
+
+    let ret = rt
+        .call::<MarketActor>(Method::ActivateDeals as u64, &RawBytes::serialize(params).unwrap())
+        .unwrap();
+    assert_eq!(ret, RawBytes::default());
+    rt.verify();
+
+    for d in deal_ids {
```

Comment:
Let's limit the number of 1-letter abbreviations to make it as readable as possible. Same goes for other methods here.